examples/scripts/kto.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2024 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Run the KTO training script with the following command with some example arguments:
+
+python examples/scripts/kto.py \
+    --per_device_train_batch_size 8 \
+    --gradient_accumulation_steps 2 \
+    --learning_rate 1e-4 \
+    --max_steps 1000 \
+    --report_to "wandb" \
+    --gradient_checkpointing True  \
+    --output_dir="./test" \
+    --use_peft True \
+    --peft_lora_r 64 \
+    --peft_lora_alpha 16 \
+    --evaluation_strategy "steps" \
+    --logging_first_step True \
+    --logging_steps 10 \
+    --eval_steps 500
+"""
+
+from dataclasses import dataclass, field
+from typing import Optional
+
+from datasets import Dataset, load_dataset
+from peft import LoraConfig
+from transformers import AutoModelForCausalLM, AutoTokenizer, HfArgumentParser
+
+from trl import KTOConfig, KTOTrainer
+
+
+# Define and parse arguments.
+@dataclass
+class ScriptArguments:
+    """
+    The arguments for the KTO training script.
+    """
+
+    # training parameters
+    model_name_or_path: Optional[str] = field(default="gpt2", metadata={"help": "the model name"})
```

Comment:
All these model init parameters are included in `ModelConfig`, so let's use that for consistency with our other scripts